File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    string answer = "";
    int index1 =0 , index2 = 0;
    bool found = true;
    for (int i = 0; i < goal.size(); i++)
    {
        auto target = goal[i];
        if (index1 <  cards1.size() && cards1[index1] == target)
        {
            index1++;
        }
        else if (index2 <  cards2.size() && cards2[index2] == target) 
        {
            index2++;
        }
        else 
        {
            found = false;
            break;
        }
    }
    
    answer = found ? "Yes" : "No";
    
    return answer;
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cs
using System;

public class Solution {
    public string solution(string[] cards1, string[] cards2, string[] goal) {
        string answer = "";
        
        int index1=0;
        int index2=0;
        bool found = true;
        
        for(int i = 0; i < goal.Length; i++)
        {
            if (index1 < cards1.Length && cards1[index1] == goal[i])
            {
                index1++;
            }
            else if (index2 < cards2.Length && cards2[index2] == goal[i])
            {
                index2++;
            }
            else 
            {
                found = false;
                break;
            }
        }
        return answer = found ? "Yes" : "No";
    }
}